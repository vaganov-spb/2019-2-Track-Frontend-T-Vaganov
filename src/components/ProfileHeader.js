/* eslint-disable react/no-unused-state */
import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { UserContext } from './UserContext';
import { HeaderTop } from './ListOfChatsHeader';
import { ArrowLeft } from './ChatHeaderInfo';
import headerStyles from '../styles/ProfileHeader.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function ProfileHeader(props) {
	const user = useContext(UserContext);

	function SaveToLS() {
		localStorage.setItem('profile', JSON.stringify(user));
	}
	
	return (
		<div className={headerStyles.main_head}>
			<HeaderTop/>
			<div className={headerStyles.message_header_bottom}>
				<ArrowLeft>
					<Link style={{color: 'var(--white)',}} to="/">
						<i className="fa fa-arrow-left" />
					</Link>
				</ArrowLeft>
				<div className={headerStyles.center}>
					<span className={headerStyles.edit_profile}>
                        Edit Profile
					</span>
				</div>     
				<div className={headerStyles.mark}>
					<img className={headerStyles.mark_img} onClick={SaveToLS}src="https://icon-library.net/images/check-icon-png/check-icon-png-13.jpg" alt=""/>
				</div>  
			</div>
		</div>
	);
}