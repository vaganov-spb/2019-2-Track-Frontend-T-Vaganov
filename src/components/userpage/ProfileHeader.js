/* eslint-disable react/no-unused-state */
import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { HeaderTop } from '../chatslist/ListOfChatsHeader';
import { ArrowLeft } from '../chatform/ChatHeaderInfo';
import headerStyles from '../../styles/ProfileHeader.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

function ProfileHeader(props) {
	function SaveToLS() {
		localStorage.setItem('profile', JSON.stringify(props.user));
	}
	
	return (
		<div className={headerStyles.main_head}>
			<HeaderTop/>
			<div className={headerStyles.message_header_bottom}>
				<ArrowLeft>
					<Link style={{color: 'var(--white)',}} to="/">
						<i className="fa fa-arrow-left" />
					</Link>
				</ArrowLeft>
				<div className={headerStyles.center}>
					<span className={headerStyles.edit_profile}>
                        Edit Profile
					</span>
				</div>     
				<div className={headerStyles.mark}>
					<img className={headerStyles.mark_img} onClick={SaveToLS}src="https://icon-library.net/images/check-icon-png/check-icon-png-13.jpg" alt=""/>
				</div>  
			</div>
		</div>
	);
}

const mapStateToProps = (state) => ({
	user: state.userinfo.info,
	error: state.userinfo.error,
	isLoaded: state.userinfo.isLoaded,
});

export default connect(mapStateToProps)(ProfileHeader);