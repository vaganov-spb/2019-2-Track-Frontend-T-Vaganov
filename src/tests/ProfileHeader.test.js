import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import ProfileHeader from '../components/userpage/ProfileHeader';

const mockStore = configureStore();

describe('ProfileHeader renders and act correctly', () => {
	let store;
	let component;
	let originalLocalStorage;

	beforeAll(()=>{
		originalLocalStorage = window.localStorage;
		delete window.localStorage;
		Object.defineProperty(window, 'localStorage', {
			writable: true,
			value: {
				getItem: jest.fn().mockName('getItem'),
				setItem: jest.fn().mockName('setItem')
			}
		});
	});

	beforeEach(() => {
		localStorage.getItem.mockClear();
		localStorage.setItem.mockClear();

		store = mockStore({
			userinfo: {
				info: {
					Name: 'Borat',
					Nick: 'Sagdiev',
					Bio: 'Pasha Technik'
				},
				isLoaded: true,
				error: null, 
			}
		});
        
		component = renderer.create(
			<Provider store={store}>
				<MemoryRouter>
					<ProfileHeader/>
				</MemoryRouter>  
			</Provider>
		);
	});
    
	afterAll(()=>{
		Object.defineProperty(window, 'localStorage', {writable: true, value: originalLocalStorage});
	});

	it ('should render normally', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
 
	it ('should act normally', () => {
		const saveLS = component.root.findByProps({className: 'mark_img'});
		saveLS.props.onClick();
 
		expect(localStorage.setItem).toHaveBeenCalled();
		expect(localStorage.setItem).toHaveBeenCalledWith('profile', '{"Name":"Borat","Nick":"Sagdiev","Bio":"Pasha Technik"}');
	});
   
});
    


