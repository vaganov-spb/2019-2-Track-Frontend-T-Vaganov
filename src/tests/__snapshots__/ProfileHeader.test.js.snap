// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHeader renders and act correctly should render normally 1`] = `
<div
  className="main_head"
>
  <div
    className="icons"
  >
    <div
      className="top_icons"
    >
      <div
        className="place_top_icons"
      >
        <span
          className="fa_icons"
        >
          <i
            className="fa fa-circle"
          />
        </span>
        <span
          className="fa_icons"
        >
          <i
            className="fa fa-battery-three-quarters"
          />
        </span>
        <span
          className="fa_icons"
        >
          <i
            className="fa fa-apple"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="message_header_bottom"
  >
    <div
      className="arrow"
    >
      <span
        className="left_icon"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "var(--white)",
            }
          }
        >
          <i
            className="fa fa-arrow-left"
          />
        </a>
      </span>
    </div>
    <div
      className="center"
    >
      <span
        className="edit_profile"
      >
        Edit Profile
      </span>
    </div>
    <div
      className="mark"
    >
      <img
        alt=""
        className="mark_img"
        onClick={[Function]}
        src="https://icon-library.net/images/check-icon-png/check-icon-png-13.jpg"
      />
    </div>
  </div>
</div>
`;
